packages/map/src/CanvasPins.ts

```diff
@@ -0,0 +1,351 @@
+    arrowWidth(_: number): CanvasPins;
+    bottomRightX(): number;
+    bottomRightX(_: number): CanvasPins;
+    bottomRightY(): number;
+    bottomRightY(_: number): CanvasPins;
+    topLeftX(): number;
+    topLeftX(_: number): CanvasPins;
+    topLeftY(): number;
+    topLeftY(_: number): CanvasPins;
+    pinHeight(): number;
+    pinHeight(_: number): CanvasPins;
+    pinWidth(): number;
+    pinWidth(_: number): CanvasPins;
+    pinFontFamily(): string;
+    pinFontFamily(_: string): CanvasPins;
+    pinFontSize(): number;
+    pinFontSize(_: number): CanvasPins;
+    bottomRightX_exists(): boolean;
+    bottomRightY_exists(): boolean;
+    topLeftX_exists(): boolean;
+    topLeftY_exists(): boolean;
+}
+
+CanvasPins.prototype.publish("allCircles", false, "boolean", "allCircles", null, { tags: ["Basic"], optional: true });
+CanvasPins.prototype.publish("showQuadtree", false, "boolean", "showQuadtree", null, { tags: ["Basic"], optional: true });
+CanvasPins.prototype.publish("useAveragePos", false, "boolean", "useAveragePos", null, { tags: ["Basic"], optional: true });
+CanvasPins.prototype.publish("shrinkFontToPin", true, "boolean", "shrinkFontToPin", null, { tags: ["Basic"], optional: true });
+CanvasPins.prototype.publish("enableClustering", true, "boolean", "enableClustering", null, { tags: ["Basic"], optional: true });
+CanvasPins.prototype.publish("useWeightedRadius", false, "boolean", "useWeightedRadius", null, { tags: ["Basic"], optional: true });
+CanvasPins.prototype.publish("radiusWeightMult", 0.5, "number", "radiusWeightMult", null, { tags: ["Basic"], optional: true });
+CanvasPins.prototype.publish("searchRectMult", 3, "number", "searchRectMult", null, { tags: ["Basic"], optional: true });
+CanvasPins.prototype.publish("bottomRightX", null, "number", "Bottom right x-value", null, { tags: ["Basic"], optional: true });
+CanvasPins.prototype.publish("bottomRightY", null, "number", "Bottom right y-value", null, { tags: ["Basic"], optional: true });
+CanvasPins.prototype.publish("topLeftX", null, "number", "Top left x-value", null, { tags: ["Basic"], optional: true });
+CanvasPins.prototype.publish("topLeftY", null, "number", "Top left y-value", null, { tags: ["Basic"], optional: true });
+
+CanvasPins.prototype.publish("pinHeight", 20, "number", "pinHeight", null, { tags: ["Basic"], optional: true });
+CanvasPins.prototype.publish("pinWidth", 20, "number", "pinWidth", null, { tags: ["Basic"], optional: true });
+CanvasPins.prototype.publish("pinFontFamily", "Arial", "string", "pinFontFamily", null, { tags: ["Basic"], optional: true });
+CanvasPins.prototype.publish("pinFontSize", 14, "number", "pinFontSize", null, { tags: ["Basic"], optional: true });
+CanvasPins.prototype.publish("arrowHeight", 8, "number", "arrowHeight", null, { tags: ["Basic"], optional: true });
+CanvasPins.prototype.publish("arrowWidth", 8, "number", "arrowWidth", null, { tags: ["Basic"], optional: true });
+
+export class Quadtree {
```

Comment:
No need to export its specific to your pins implementation - also I would move it to above the other class (as in general we prefer to declare the class before it is used - just a convention).